### PR TITLE
Adding support for experimental Custom CA integration using Kubernetes CSR API

### PR DESCRIPTION
**Pull request by shankgan**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Review by myidpt — APPROVED** [review]

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    {{< /text >}}
+
+## Part 2: Using Custom CA
+
+This assumes that the custom CA implements a controller that has the necessary permissions to read and sign Kubernetes CSR Requests. Refer (<https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/>) for more details. Note that the steps below are dependent on an external-source and may change.
+
+### Deploy Custom CA controller in the Kubernetes cluster
+
+1. For this example, we use an open-source Certificate Authority implemented here
+   <https://github.com/cert-manager/signer-ca>. This code builds a controller that reads the CSR resources creates on the Kubernetes cluster and creates certificates using local keys. Follow the instructions on the page to
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:108]:
```suggestion
1. For this example, we use an [open-source Certificate Authority implementation](https://github.com/cert-manager/signer-ca).
    This code builds a controller that reads the CSR resources on the Kubernetes cluster and creates certificates using local keys. Follow the instructions on the page to:
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+   1. Build the Certificate-Controller docker image
+   1. Upload it to a Docker Registry and
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:110]:
```suggestion
   1. Upload the image to a Docker Registry
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+   1. Generate the Kubernetes manifest to deploy the same.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:111]:
```suggestion
   1. Generate the Kubernetes manifest to deploy it
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+1. Deploy the Kubernetes manifest generated in the step above on your local cluster in the signer-ca-system namespace
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:113]:
```suggestion
1. Deploy the Kubernetes manifest generated in the previous step on your local cluster in the signer-ca-system namespace.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ kubectl apply -f local-ca.yaml
+    {{< /text >}}
+
+   Ensure that all the services are running
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:119]:
```suggestion
   Ensure that all the services are running.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ kubectl get services -n signer-ca-system
+      NAME                                           TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+      signer-ca-controller-manager-metrics-service   ClusterIP   10.8.9.25    none        8443/TCP   72s
+    {{< /text >}}
+
+1. Get the public key of the CA. This is encoded in the secret "signer-ca-*" in the signer-ca-system namespace
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:127]:
```suggestion
1. Get the public key of the CA. This is encoded in the secret "signer-ca-*" in the signer-ca-system namespace.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ kubectl get secrets signer-ca-5hff5h74hm -o json
+    {{< /text >}}
+
+   The `tls.crt` field contains the base64 encoded public key file. Record this for future use.
+
+### Load the CA root certificate into a secret that istiod can access
+
+1. Load the secret into the istiod namespace
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:137]:
```suggestion
1. Load the secret into the istiod namespace.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+      metadata:
+        name: external-ca-cert
+        namespace: istio-system
+      data:
+      root-cert.pem: <tls.cert from the step above>
+      EOF
+    $ kubectl apply -f external-ca-secret.yaml
+    {{< /text >}}
+
+    This step is necessary for Istio to verify that the workload certificates have been signed by the correct certificate authority and adds the root-cert to the trust bundle for mTLS to work.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:152]:
```suggestion
    This step is necessary for Istio to verify that the workload certificates have been signed by the correct certificate authority and to add the root-cert to the trust bundle for mTLS to work.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+### Deploying Istio
+
+1. Deploy Istio on the cluster using `istioctl` with the feature configuration applied.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:156]:
```suggestion
1. Deploy Istio on the cluster using `istioctl` with the following configuration.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                  value: |
+                    name: external-ca-cert
+                    secret:
+                      secretName: external-ca-cert
+                      optional: true
+    EOF
+    $ istioctl install --set profile=demo -f ./istio.yaml
+    {{< /text >}}
+
+1. Deploy the `bookinfo` sample application in the bookinfo namespace. Ensure that the instructions below are executed in the Istio directory.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:211]:
```suggestion
1. Deploy the `bookinfo` sample application in the bookinfo namespace.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ kubectl create ns bookinfo
+    $ kubectl apply -f <(istioctl kube-inject -f samples/bookinfo/platform/kube/bookinfo.yaml) -n bookinfo
+    {{< /text >}}
+
+### Verify that Custom CA certificates installed are correct
+
+When the workloads are deployed above, they send CSR Requests to Istiod which forwards them to the Kubernetes CA for signing. If all goes well, the signed certificates are send back to the workloads where they are installed. To verify that they have indeed been signed by the Kubernetes CA, we need to first extract the signed certificates.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:220]:
```suggestion
When the workloads are deployed, above, they send CSR Requests to Istiod which forwards them to the Kubernetes CA for signing. If all goes well, the signed certificates are sent back to the workloads where they are then installed. To verify that they have indeed been signed by the Kubernetes CA, you need to first extract the signed certificates.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+
+1. Dump all pods running in the namespace.
+
+    {{< text bash >}}
+    $ kubectl get pods -n bookinfo
+    {{< /text >}}
+
+   Pick any of the running pods for the next step
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:228]:
```suggestion
   Pick any of the running pods for the next step.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+
+1. Get the certificate chain and CA root certificate used by the Istio proxies for mTLS
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:230]:
```suggestion
1. Get the certificate chain and CA root certificate used by the Istio proxies for mTLS.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 14]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ istioctl pc secret $(pod-name) -o json > proxy_secret
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:233]:
```suggestion
    $ istioctl pc secret <pod-name> -o json > proxy_secret
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 15]
```diff
@@ -0,0 +1,255 @@
...
+    {{< /text >}}
+
+   The `proxy_secret` json file contains the CA root certificate for mTLS in the `trustedCA` field. Note that this certificate is base64 encoded.
+
+1. Compare the CA root certificate obtained in the step above with "root-cert.pem" value in external-ca-cert. These two should be the same.
+
+1. (Optional) Follow the rest of the steps mentioned in the bookinfo example (<https://istio.io/latest/docs/examples/bookinfo/>) to ensure that communication between services works fine.
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:240]:
```suggestion
1. (Optional) Follow the rest of the steps in the [bookinfo example](/docs/examples/bookinfo/) to ensure that communication between services is working as expected.
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 16]
```diff
@@ -0,0 +1,255 @@
...
+
+### Cleanup Part 2
+
+* To remove the `istio-system` and `bookinfo` namespace:
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:244]:
```suggestion
* Remove the `istio-system` and `bookinfo` namespaces:
```

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 17]
```diff
@@ -0,0 +1,255 @@
...
+
+    {{< text bash >}}
+    $ kubectl delete ns istio-system
+    $ kubectl delete ns bookinfo
+    {{< /text >}}
+
+## Reason to use this feature
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:251]:
```suggestion
## Reasons to use this feature
```

**Review by frankbu — CHANGES_REQUESTED** [review]

**Comment by shankgan** [comment]
Thanks a ton @frankbu  for reviewing this document. Have incorporated all your suggestions.

**Comment by frankbu** [comment]
@googlebot I consent

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 18]
```diff
@@ -0,0 +1,267 @@
+---
+title: Istio Custom Certificate Authority Integration using Kubernetes CSR API [experimental]
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:2]: Title is very long. How about shortening:

```suggestion
title: Custom CA Integration using Kubernetes CSR [experimental]
```

Comment (shankgan) [reply content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md]: done

**Comment by shankgan** [comment]
@googlebot I consent

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md [hunk 19]
```diff
@@ -0,0 +1,267 @@
...
+description: Shows how to use a Custom Certificate Authority (that integrates with the Kubernetes CSR API) to provision Istio workload certificates.
+weight: 90
```
Comment (frankbu) [inline content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:4]: This controls where in the list of documents under the same folder, this one should go. I think it should be last, so higher than the other highest one (90).
```suggestion
weight: 100
```

Comment (shankgan) [reply content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md:4]: Agreed

**Comment by shankgan** [comment]
@frankbu - Is there a reason you removed the Istio 1.8 cherry-pick label?

File: content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md (hunk 18 above)
Comment (frankbu) [reply content/en/docs/tasks/security/cert-management/custom-ca-k8s/index.md]: You'll need to run `make gen` again, because of the doc title change

**Comment by frankbu** [comment]
> @frankbu - Is there a reason you removed the Istio 1.8 cherry-pick label?

Doc PRs don't need cherrypick labels until after the release date.

**Review by frankbu — APPROVED** [review]

**Pull request merged**